packages/block-editor/src/components/block-popover/index.js

```diff
@@ -88,6 +124,45 @@ function BlockPopover(
 
 		return {
 			getBoundingClientRect() {
+				let postRootElement;
+				// The zoom out view has a vertical block toolbar that should always
+				// be on the edge of the canvas, aligned to the top of the currently
+				// selected section. This condition changes the anchor of the toolbar
+				// to the section instead of the block to handle blocks that are
+				// not full width and nested blocks to keep section height.
+				if ( isZoomOut && isSectionSelected ) {
+					// if the rootSectionElement is undefined then we need to recurse up the DOM tree
+					// to find the element with  wp-block-post-content classname
+					if ( ! rootSectionElement ) {
+						postRootElement =
+							selectedElement.closest( '.is-root-container' );
+					}
```

Comment:
@draganescu Potentially brittle but it works. We can use the root container element as the anchor point. 

The concern would be that this is an API that could change and one that is set by a package outside of the block editor. Therefore it's probably not resilient enough.

Reply:
Yes I also tried to hook into `.is-root-container` ... but it doesn't seem like the best option.